database/src/main/java/com/firebase/ui/database/FirebaseAdapter.java

```diff
@@ -1,26 +1,25 @@
 package com.firebase.ui.database;
 
-import android.arch.lifecycle.LifecycleObserver;
+import android.support.annotation.NonNull;
 import android.support.annotation.RestrictTo;
 
+import com.firebase.ui.common.Adapter;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
 import com.google.firebase.database.DatabaseReference;
 
 @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
-interface FirebaseAdapter<T> extends ChangeEventListener, LifecycleObserver {
+interface FirebaseAdapter<T> extends Adapter<ObservableSnapshotArray<T>,
+        DataSnapshot, DatabaseError,
+        T>, ChangeEventListener {
     /**
-     * If you need to do some setup before the adapter starts listening for change events in the
-     * database, do so it here and then call {@code super.startListening()}.
```

Comment:
The documentation in this class was all wrong! 😱 `startListening` is no longer called in the constructor, `stopListening` doesn't necessarily clear the array, and we were missing documentation for the other methods.

Reply:
Thanks for catching this!